src/Functions/parseTimeDelta.cpp

```diff
@@ -0,0 +1,243 @@
+                {
+                    res_data.emplace_back(0);
+                    continue;
+                }
+
+                while (fast <= last_pos)
+                {
+                    ScanIgnoreCharacters(str, fast, last_pos);
+                    slow = fast;
+
+                    /// scan unsigned integer
+                    if (!ScanUnsignedInteger(str, fast, last_pos))
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " find number failed: " + String(str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+
+                    /// if there is a '.', then scan another integer to get a float number
+                    if (fast <= last_pos && str[fast] == '.')
+                    {
+                        fast++;
+                        if (!ScanUnsignedInteger(str, fast, last_pos))
+                        {
+                            throw Exception(
+                                "bad auguments invalid expression for function " + getName()
+                                    + " find number after '.' failed: " + String(str) + ".",
+                                ErrorCodes::BAD_ARGUMENTS);
+                        }
+                    }
+
+                    /// convert float/integer string to float
+                    Float64 base = 0;
+                    std::string_view base_str = str.substr(slow, fast - slow);
+                    auto value = boost::convert<Float64>(base_str, boost::cnv::strtol());
+                    if (!value.has_value())
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName()
+                                + " convert string to float64 failed:  " + String(base_str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+                    base = value.get();
+
+                    ScanIgnoreCharacters(str, fast, last_pos);
+                    slow = fast;
+
+                    /// scan a unit
+                    if (!ScanUnit(str, fast, last_pos))
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " find unit failed: " + String(str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+
+                    /// get unit number
+                    std::string_view unit = str.substr(slow, fast - slow);
+                    auto iter = time_unit_to_float.find(unit);
+                    if (iter == time_unit_to_float.end()) /// not find unit
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " parse unit failed: " + String(iter->first)
```

Comment:
Using `iter->first` leads to segfault, because `iter` is invalid here. You can run query like `SELECT parseTimeDelta('2 minn')` and it will crash server. We should use `unit` here